Set sync setting in config automatically (#18)
stolostron/gatekeeper-operator: docs/upgrading-gatekeeper.md

## 10. Update unit and e2e tests

> The gatekeeper package version should be updated with the matched Gatekeeper version in go.mod.

Reviewer: This line should be back to what it was before.

> ## 11. Updating gatekeeper 

Reviewer: ```suggestion
## 11. Updating the gatekeeper Go module 
```

> ### Update gatekeeper package according to gatekeeper version 
> 
> The gatekeeper package version should be updated with the matched Gatekeeper version in go.mod.

Reviewer: ```suggestion
The Gatekeeper Go module version in `go.mod` should be updated to match the Gatekeeper version being deployed by the operator.
```

> ```
> require github.com/open-policy-agent/gatekeeper/v3 v3.13.4
> ```

Reviewer: untested
```suggestion
sed -i "s/require github.com/open-policy-agent/gatekeeper/v3  .*/require github.com/open-policy-agent/gatekeeper/v3 ${GATEKEEPER_VERSION}/" go.mod
go mod tidy
```

Make sure to add or modify any unit and e2e tests as a result of any operator
controller changes.

## 11. Updating gatekeeper 

### Update gatekeeper package according to gatekeeper version 

> The gatekeeper package version should be updated with the matched the gatekeeper version when gatekeeper version is updated in go.mod.

Reviewer: ```suggestion
The gatekeeper package version should be updated with the matched Gatekeeper version in go.mod.
```
